packages/jellyfish-api-core/__tests__/category/loan/updateLoanScheme.test.ts

```diff
@@ -0,0 +1,185 @@
+  })
+
+  it('should updateLoanScheme at activateAfterBlock which is block 120', async () => {
+    // Wait for block 110
+    await testing.container.waitForBlockHeight(110)
+
+    // To update at block 120
+    const loanSchemeId = await testing.rpc.loan.updateLoanScheme(300, new BigNumber(3.5), { id: 'scheme1', activateAfterBlock: 120 })
+    expect(typeof loanSchemeId).toStrictEqual('string')
+    expect(loanSchemeId.length).toStrictEqual(64)
+    await testing.generate(1)
+
+    // Shouldn't update at block 111
+    {
+      const data = await testing.container.call('listloanschemes')
+      const result = data.filter((d: { id: string }) => d.id === 'scheme1')
+      expect(result.length).toStrictEqual(1)
+      expect(result[0]).toStrictEqual(
+        {
+          default: false,
+          id: 'scheme1',
+          interestrate: 2.5,
+          mincolratio: 200
+        }
+      )
+    }
+
+    // Wait for block 120
+    await testing.container.waitForBlockHeight(120)
+
+    // Should update at block 120
+    {
+      const data = await testing.container.call('listloanschemes')
+      const result = data.filter((d: { id: string }) => d.id === 'scheme1')
+      expect(result.length).toStrictEqual(1)
+      expect(result[0]).toStrictEqual(
+        {
+          default: false,
+          id: 'scheme1',
+          interestrate: 3.5,
+          mincolratio: 300
+        }
+      )
+    }
+  })
+
+  it('should not updateLoanScheme if activateAfterBlock is less than current block', async () => {
+    // Wait for block 130
+    await testing.container.waitForBlockHeight(130)
+
+    // Attempt to updateLoanScheme at block 129
+    const promise = testing.rpc.loan.updateLoanScheme(300, new BigNumber(3.5), { id: 'scheme1', activateAfterBlock: 129 })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nUpdate height below current block height, set future height\', code: -32600, method: updateloanscheme')
+  })
+
+  it('should not updateLoanScheme if same minColRatio and interestRate pending loan scheme created before', async () => {
+    // Wait for block 140
+    await testing.container.waitForBlockHeight(140)
+
+    await testing.container.call('createloanscheme', [300, new BigNumber(3.5), 'scheme2'])
+    await testing.generate(1)
+
+    // To update scheme on later block
+    await testing.rpc.loan.updateLoanScheme(400, new BigNumber(4.5), { id: 'scheme2', activateAfterBlock: 150 })
+    await testing.generate(1)
+
+    // Attempt to update same minColRatio and interestRate as the pending scheme
+    const promise = testing.rpc.loan.updateLoanScheme(400, new BigNumber(4.5), { id: 'scheme1' })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSchemeTx execution failed:\nLoan scheme scheme2 with same interestrate and mincolratio pending on block 150\', code: -32600, method: updateloanscheme')
+
+    // Update at block 150
+    await testing.container.waitForBlockHeight(150)
```

Comment:
Not needed

Reply:
If don't do this, the code for this test unit maybe run in the next `it`